orchagent/aclorch.cpp

```diff
@@ -301,6 +301,12 @@ const set<sai_acl_action_type_t>& AclTableType::getActions() const
     return m_aclAcitons;
 }
 
+bool AclTableType::addAction(sai_acl_action_type_t action)
```

Comment:
```AclTableType``` was designed to be an immutable data structure, so you cannot change it after it has been created. This is due to the fact that matches, actions, bind point types are CREATE_ONLY SAI attributes. Adding mutable methods breaks this invariant. An object of type ```AclTableType``` may have been used to create tables already and changing the definition of a table type may cause divergence between software and hardware state.
Instead, a builder ```AclTableTypeBuilder``` is used to create ```AclTableType```s.
Why not adding neccessary actions in ```initDefaultTableTypes``` - https://github.com/Azure/sonic-swss/blob/master/orchagent/aclorch.cpp#L2895 ?

Reply:
> `AclTableType` was designed to be an immutable data structure, so you cannot change it after it has been created. This is due to the fact that matches, actions, bind point types are CREATE_ONLY SAI attributes. Adding mutable methods breaks this invariant. An object of type `AclTableType` may have been used to create tables already and changing the definition of a table type may cause divergence between software and hardware state. Instead, a builder `AclTableTypeBuilder` is used to create `AclTableType`s. Why not adding neccessary actions in `initDefaultTableTypes` - https://github.com/Azure/sonic-swss/blob/master/orchagent/aclorch.cpp#L2895 ?

We also saw similar issue on broadcom paltform. Please see https://github.com/Azure/sonic-buildimage/issues/10425.
We may have two options to workaround this issue
1. Hardcode `is_mandatory` to `False` for the general ACL type, such as `L3`, `L3V6`
2. Add default action list for these ACL tables
What do you think?

Reply:
Ok, I see, so the issue may be due to SAI telling that action list is mandatory even if it is not (because it worked without actions before ACL table types changes). In this case, hardcoding is_mandatory to be False for the types we know should work without explicitly passing action list should be a simpler workaround. Put a comment that this hardcode needs to be removed once SAI is fixed.

Reply:
Based on our communication with broadcom,  action list is mandatory when creating mirror acl table. Otherwise,  SAI acl table creation will fail. Can you elaborate how would hardcoding is_mandatory to be False work around the SAI limitation?

Reply:
If we only bypass the above check, aclorch attempts to create mirror acl table without passing mandatory actions and the corresponding SAI call will fail on platforms like broadcom that requires actions in acl table creation.

Reply:
> Based on our communication with broadcom, action list is mandatory when creating mirror acl table. Otherwise, SAI acl table creation will fail. Can you elaborate how would hardcoding is_mandatory to be False work around the SAI limitation?

@ysmanman This means mirroring never worked on some platforms or wasn't tested? In 201911 for all table types there is no SAI_ACL_TABLE_ATTR_ACL_ACTION_TYPE_LIST https://github.com/Azure/sonic-swss/blob/201911/orchagent/aclorch.cpp#L1516.

Reply:
@stepanblyschak Mirroring probably worked in fixed box before. My PR is to get mirroring work in broadcom VOQ chassis.

Reply:
> 

Thanks for the update. Could you clarify why SAI call will fail if we bypass this check? Before this change, there is no check for action list existance. Is it a new feature of SAI?

Reply:
@bingwang-ms @stepanblyschak Let me provide some context of this PR. We were testing mirroring in broadcom DNX or VOQ chassis with broadcom SAI 5.2 and observed orchagent failed to add ACL rule in mirroring ACL table.  The failure is because the ACL actions is not provided at the time of mirror ACL table creation.  Broadcom DNX  devices need ACL action list at the time of ACL table creation. To address this issue,  this PR adds required ACL actions to mirroring type ACL table if the action list is mandatory on table creation.

Reply:
Thanks @ysmanman .
I saw similar issues when testing on Broadcom DNX devices. Please see https://github.com/Azure/sonic-buildimage/issues/10425.
Per your clarification, the action_list for ACL table is a must when we creating ACL table now? If that's the case, my by-pass solution will not work. The solution in this PR makes sense. We have to do similar change for other ACL table types (L3, L3V6, and etc.)